File: Competitions/UC/Berkeley-2024/contest/contest/reservoir/templates/reservoir_template.java
import java.util.Scanner;

class Solution {
    /**
     * Return the maximum number of islands.
     *
     * N: number of rows
     * M: number of columns
     * G: grid of heights
     */
    public static int solve(int N, int M, int[][] G) {
        return 0;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        int T = sc.nextInt(); // Number of test cases
        for (int t = 0; t < T; t++) {
            int N = sc.nextInt(); // Number of rows
            int M = sc.nextInt(); // Number of columns
            int[][] G = new int[N][M];
            
            // Reading the grid
            for (int i = 0; i < N; i++)
                for (int j = 0; j < M; j++)
                    G[i][j] = sc.nextInt();
            
            // Call the solve function and print the result
            System.out.println(solve(N, M, G));
        }
        
        sc.close();
    }
}

File: Competitions/UC/Berkeley-2024/contest/contest/reservoir/templates/reservoir_template.cpp
#include <iostream>
#include <vector>

using namespace std;

/**
 * Return the maximum number of islands.
 *
 * N: number of rows
 * M: number of columns
 * G: grid of heights
 */
int solve(int N, int M, vector<vector<int>> &G) {
    return 0;
}

int main() {
    int T;
    cin >> T;
    for (int t = 0; t < T; t++) {
        int N, M;
        cin >> N >> M;
        vector<vector<int>> G(N, vector<int>(M));
        for (int i = 0; i < N; i++)
            for (int j = 0; j < M; j++)
                cin >> G[i][j];
        cout << solve(N, M, G) << '\n';
    }
}
